lib/protobuf/enum.rb

```diff
@@ -198,7 +206,7 @@ def self.name_for_tag(tag)
     # Returns a boolean.
     #
     def self.valid_tag?(tag)
-      tag.respond_to?(:to_i) && all_tags.include?(tag.to_i)
+      tag.respond_to?(:to_i) && mapped_enums.key?(tag.to_i)
```

Comment:
It looks like `valid_tag?` is a public API that's not used internally, so it wouldn't speed up performance in (our) case. It's probably true that a lot of cases could be replaced with rescuing `NoMethodError` if it's rare that invalid data is passed. Would leave that up to the owners if they're interested before I go dig through that.